parl/remote/scripts.py

```diff
@@ -0,0 +1,123 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import locale
+import os
+import subprocess
+import threading
+import warnings
+from multiprocessing import Process
+
+# A flag to mark if parl is started from a command line
+os.environ['XPARL'] = 'True'
+
+# Solve `Click will abort further execution because Python 3 was configured
+# to use ASCII as encoding for the environment` error.
+locale.setlocale(locale.LC_ALL,"en_US.UTF-8")
+
+warnings.simplefilter("ignore", ResourceWarning)
+
+
+def is_port_in_use(port):
+    """ Check if a port is used.
+
+    0: The port is used.
+    else: The port is not used.
+    """
```

Comment:
Returns:
     True if the port is not available. Otherwise, this port can be used for connection.